docs/TMC_Drivers.md

```diff
@@ -70,124 +62,259 @@ of the motor, so the TMC cannot detect stalls anymore. It is advised
+movement commands (eg, `G1`). Klipper will not have a correct
+understanding of the carriage position and a move command may cause
+undesirable and confusing results.
+
+### Find lowest sensitivity that homes with one touch
+
+When homing with the found `maximum_sensitivity` value, the axis
+should move to the end of the rail and stop with a "single touch" -
+that is, there should not be a "clicking" or "banging" sound. (If
+there is a banging or clicking sound at maximum_sensitivity then the
+homing_speed may be too low, the driver current may be too low, or
+sensorless homing may not be a good choice for the axis.)
+
+The next step is to again continually move the carriage to a position
+near the center of the rail, decrease the sensitivity, and run the
+`SET_TMC_FIELD` `G28 X0` commands - the goal is now to find the lowest
+sensitivity that still results in the carriage successfully homing
+with a "single touch". That is, it does not "bang" or "click" when
+contacting the end of the rail. Note the found value as
+`minimum_sensitivity`.
+
+### Update printer.cfg with sensitivity value
+
+After finding `maximum_sensitivity` and `minimum_sensitivity`, use a
+calculator to obtain the recommend sensitivity as
+`minimum_sensitivity + (maximum_sensitivity - minimum_sensitivity)/3`.
+The recommended sensitivity should be in the range between the minimum
+and maximum, but slightly closer to the minimum. Round the final value
+to the nearest integer value.
+
+For tmc2209 set this in the config as `driver_SGTHRS`, for other TMC
+drivers set this in the config as `driver_SGT`.
+
+If the range between `maximum_sensitivity` and `minimum_sensitivity`
+is small (eg, less than 10) then it may result in unstable homing. A
+faster homing speed may increase the range and make the operation more
+stable.
+
+Note that if any change is made to driver current, homing speed, or a
+notable change is made to the printer hardware, then it will be
+necessary to run the tuning process again.
+
+### Using Macros when Homing
+
+After sensorless homing completes the carriage will be pressed against
+the end of the rail and the stepper will exert a force on the frame
+until the carriage is moved away. It is a good idea to create a macro
+to home the axis and immediately move the carriage away from the end
+of the rail. It is recommended to set the speed of this subsequent
+move so that it lasts at least two seconds (eg, `G1 X40 F1200`) to
+ensure the stall flag in the TMC driver is cleared after the move
+completes.
+
+It can also be useful to have that macro set the driver current before
+homing and set a new current after the carriage has moved away. This
+also allows a hold_current to be set during prints (a hold_current
+is not recommended during sensorless homing).
+
+An example macro might look like:
+```
+[gcode_macro SENSORLESS_HOME_X]
+default_parameter_HOMECUR: 0.700
+default_parameter_RUNCUR:  0.800
+default_parameter_HOLDCUR: 0.500
```

Comment:
This seems dangerous to me. duplicating something as critical as current values in a macro vs the driver config seems like a really good place to end up with bug reports reading "I changed my currents but it didn't seem to do anything"

I know its been brought up exhaustively, but since the concept of homing current seems specific to sensorless homing why not fold this behavior into the existing driver configs instead so we might avoid duplication, macro complexity and risk? To me at least the macro feels like a clever workaround to a missing feature, which is pretty cool you can do it, its just complicated and I fear makes the approach for new comers even more daunting. 

Though I am still not convinced homing specific current is needed broadly enough to justify it as a feature complicating homing logic further.

Is there a way to create a top level constant in the cfg file that could be referenced here and in the driver config? might be a simpler approach to tame the risk, though wouldn't eliminate it.